Lib/fontbakery/profiles/gdef.py

```diff
@@ -0,0 +1,116 @@
+from fontbakery.callable import check
+from fontbakery.checkrunner import FAIL, PASS, WARN
+from fontbakery.message import Message
+from fontbakery.utils import pretty_print_list
+
+# used to inform get_module_profile whether and how to create a profile
+from fontbakery.fonts_profile import profile_factory # NOQA pylint: disable=unused-import
+
+from fontTools import unicodedata
+
+@ check(
+  id = 'com.google.fonts/check/gdef_spacing_marks',
+  rationale ="""
+  Glyphs in the GDEF mark glyph class should be non-spacing.
+  Spacing glyphs in the GDEF mark glyph class may have incorrect
+  anchor positioning that was only intended for building composite glyphs
+  during design.
+  """
+)
+def com_google_fonts_check_gdef_spacing_marks(ttFont):
+  """Check mark characters are in GDEF mark glyph class)"""
+  if "GDEF" in ttFont:
+    spacing_glyphs = {name for (name, value) in ttFont["hmtx"].metrics.items()
+                      if value}
+    class_defs = ttFont["GDEF"].table.GlyphClassDef.classDefs.items()
+    mark_class_glyphs = set([name for (name, value) in class_defs if value == 3])
+    spacing_glyphs_in_mark_glyph_class = spacing_glyphs & mark_class_glyphs
+    if spacing_glyphs_in_mark_glyph_class :
+      formatted_list = "\t " +\
+        pretty_print_list(sorted(spacing_glyphs_in_mark_glyph_class),
+                          shorten=10,
+                          sep=", ")
+      yield WARN,\
+        Message('spacing glyphs in mark glyph class',
```

Comment:
Does this keyword need to be that long? Please try to come up with something shorter. Also, we tend to not use spaces. We do not have a hard rule here, but in general I have been using dashes when word separation is needed in slightly longer keywords.